### PR TITLE
feat(dashboard,icons,types,js-sdk): add providers to location UI

### PR DESCRIPTION
**Pull request by riqwan**
what:

- UI to add providers to location once the option has been created

bug found:

There is a weird overlapping back button when the modal comes up. Can be seen in the video.

https://github.com/user-attachments/assets/8ffcc648-c218-43dd-aeb4-9b098ab6de48

RESOLVES CC-231

### CODE REVIEW

File: packages/admin-next/dashboard/src/hooks/api/stock-locations.tsx [hunk 1]
```diff
@@ -175,3 +176,31 @@ export const useCreateStockLocationFulfillmentSet = (
...
+  return useMutation({
+    mutationFn: (payload) =>
+      sdk.admin.stockLocation.updateFulfillmentProviders(id, payload),
+    onSuccess: async (data, variables, context) => {
+      await queryClient.invalidateQueries({
+        queryKey: stockLocationsQueryKeys.details(),
+      })
+      await queryClient.invalidateQueries({
+        queryKey: stockLocationsQueryKeys.lists(),
+      })
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/hooks/api/stock-locations.tsx]: nit: think we can skip invalidating the lists key for this one

File: packages/admin-next/dashboard/src/routes/locations/location-detail/components/location-fulfillment-providers-section/location-fulfillment-providers-section.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      </div>
+
+      {location?.fulfillment_providers?.length ? (
+        <div className="flex flex-col gap-y-4 pt-4">
+          <div className="grid grid-cols-[28px_1fr] items-center gap-x-3">
+            <IconAvatar>
+              <HandTruck className="text-ui-fg-subtle" />
+            </IconAvatar>
+
+            <ListSummary
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/routes/locations/location-detail/components/location-fulfillment-providers-section/location-fulfillment-providers-section.tsx]: question: should the "+1 more" be displayed inline without spacing in between and right alignment?

File: packages/admin-next/dashboard/src/routes/locations/location-fulfillment-providers/components/edit-fulfillment-providers-form/edit-fulfillment-providers-form.tsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+  })
+
+  if (isError) {
+    throw error
+  }
+
+  return (
+    <RouteFocusModal.Form form={form}>
+      <div className="flex h-full flex-col overflow-hidden">
+        <RouteFocusModal.Header>
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/routes/locations/location-fulfillment-providers/components/edit-fulfillment-providers-form/edit-fulfillment-providers-form.tsx]: suggestion: believe we started moving header actions to footer. Iinstead this use ` <RouteFocusModal.Footer>` after `<RouteFocusModal.Body>`.

**Review by fPolic — COMMENTED** [review]
looks good!

**Comment by riqwan** [comment]
@fPolic just checked the designs again, it should've been a list. I've changed to that. 

Screenshot 2024-07-29 at 18 02 02

File: packages/admin-next/dashboard/src/routes/locations/location-detail/components/location-fulfillment-providers-section/location-fulfillment-providers-section.tsx (hunk 2 above)
Comment (riqwan) [reply packages/admin-next/dashboard/src/routes/locations/location-detail/components/location-fulfillment-providers-section/location-fulfillment-providers-section.tsx]: added a comment below that changes this whole thing.

**Review by olivermrbl — COMMENTED** [review]
LGTM, nice work. Will let @fPolic have a last look at it 👍

**Review by fPolic — APPROVED** [review]
LGTM

**Pull request merged**